Bugfix/ponderluminance d3 d 2283

Kuranes opened the pull request:
- Ponders luminance by solid angle
- reduce light area size after zone light merge to select area light near analytic directional point

Kuranes committed:
[DEV-7192] Ponder Lightprobes with SolidAngle

Current results are resolution dependant and not solidAngle pondered, we
fix it by pondering results with solid Angle

Kuranes committed:
[DEV-7192] output color in float

Uses the float range (0,1+), not in bitmap (0-255) range as we use float
range in shader uniform anyway when we use them

Kuranes committed:
[DEV-7192] SolidAngle ponderation

Kuranes committed:
[DEV-7192] Picks smaller lights from merged lights

Merged Lights can get too big to be a realistic approcimation of a
directionnal light.

We must take smaller area inside the selected merged
light Area to pick better approximation, using a ratio between total env
map luminance and light area size ratio

Kuranes committed:
[DEV-7192] Tightens code to standards

cedricpinson commented on ExtractLightsVariance:
```diff
@@ -6,18 +6,20 @@
 #include "Math"
 
-struct double2 {
+struct Double2 {
```
Why not using Vec2d already in Math ? same with Vec3d

Commit:
Cleans filename and factor solid angle functions

- Class name has header and implementation with upper case. Cubemap / Cubemap.cpp
- Splits header/cpp files to avoid circular deps in header
- program/executable start in camel case lower case

cedricpinson commented:
@Kuranes  It seems that extractLight is still built but could probably be removed right ? if we use the extractLights we dont need the previous one.

Kuranes commented:
True, can be removed if no "existing" code conflicts (to be checked with tests done by backend ?)

Kuranes commented:
Have maths errors on the fast path of the integral image building, on big image + "vibrant" color, as summing give great values, whereas colors themself can be small values too, have to test
- slow path (not accumulative sum in one pass, but each pixel summing whole sum "behind" him)
- reducing size of input to 1024 or 512

Kuranes committed:
Add Errors detection

Detects out of range values and flags error on output

Kuranes commented:
Added Error Detection, tried quick fix for enhancing precision, but need more work...
